auth/src/api/handlers.rs

```diff
@@ -39,6 +39,23 @@ pub(crate) async fn post_user(
     Ok(Json(user.into()))
 }
 
+pub(crate) async fn put_user_reset_key(
+    session: ReadableSession,
+    State(user_manager): State<UserManagerState>,
+    key: Option<Key>,
+) -> Result<(), Error> {
+    let account_name = match session.get::<String>("account_name") {
+        Some(account_name) => account_name.into(),
+
+        None => match key {
+            Some(key) => user_manager.get_user_by_key(key.into()).await?.name,
+            None => return Err(Error::Unauthorized),
+        },
+    };
+
+    user_manager.reset_key(account_name).await
+}
```

Comment:
How do users get their new key? Maybe this should return the key?

Reply:
Hmm ... when you login via the browser, the API key is shown. The CLI command to reset the key is via `logout`, hence the user has to login again and will see the new key.

Reply:
@chesedo If a key hijacker can get the new key, they can create an infinite loop of key regeneration, locking the account owner out. You should only be able to get the key from the web.